### ISSUE TITLE
MachineRegistration config:elemental:registration:labels are not interpolated with SMBIOS data

### ISSUE THREAD

**Issue by fgiudici** [rancher/elemental-operator#161]
Labels meant for the MachineInventory created during machine registration to the Elemental Operator can be specified in two places:
- `MachineRegistration spec:machineInventoryLabels`
- `MachineRegistration spec:config:elemental:registration:labels` (these can be even passed directly as parameters to the  `elemental-register` binary)

Starting from Elemental Operator v0.5.0, labels set in the `machineInventoryLabels`  are interpolated with SMBIOS data. Labels from `config:elemental:registration:labels` instead are not filled with SMBIOS data.

**Comment by Itxaka** [comment]
as I comented in the PR, I believe the label stuff should be dropped.

Now I can see where that comes from, the elemental.registration.labels is used for the client to store the label config that is gonna be pushed to the server. When we merged the config from the registration client back into the operator and reused it we got that.

But the label stuff has and should always be in one place, especially for the registration. IMHO, label override should be dropped from the client and not used. Source of truth should be the registration object in the server only
